music_assistant/server/controllers/player_queues.py

```diff
@@ -345,6 +359,10 @@ async def play_media(
                 else:
                     media_item = item
 
+                # Save requested media item to play on the queue so we can use it as a source
+                # for Don't stop the music
+                queue.played_media_item = media_item
```

Comment:
So if I play a few tracks, it will be based on the last track of the list?

Reply:
Yep, we could also change it to be a `list` so that it grabs all the tracks that were added.

Reply:
yeah maybe a length limited list of max 10 items or so

Reply:
Out of curiosity, would this theoretically resolve my issue with the current implementation where it pulls the same 20(or so) songs from Tidal into the playlist. So instead of pulling songs based on the song the radio station was started on, it would pull a list of songs based on say the most recently played 5-10 songs?

Reply:
No, radio mode bases the radio based on the 'radio media item' (item used when the user clicked 'start radio'). However, I rewrote the logic a few beta releases ago, so that it uses individual songs to search for 'similar songs' instead of an album or an artist. Did you test if this behaviour still occurs with Tidal in the latest beta?